server/src/main/java/org/opensearch/repositories/Repository.java

```diff
@@ -321,6 +322,7 @@ void cloneShardSnapshot(
         SnapshotId target,
         RepositoryShardId shardId,
         @Nullable String shardGeneration,
+        RemoteStoreLockManagerFactory remoteStoreLockManagerFactory,
```

Comment:
Note this might be a breaking change for 2.x. How do you intend to solve this?

Reply:
Now, we have another method for cloning shards of remote enabled indices - `cloneRemoteStoreIndexShardSnapshot` with default implementation to throw `UnsupportedOperationException`. This way, the signature of existing method doesn't have to change.